lib/delayed/backend/active_record.rb

```diff
@@ -138,23 +138,35 @@ def self.jobs_count(flavor,
         # or opts[:flavor] = <some flavor> to perform on all jobs of that flavor
         def self.bulk_update(action, opts)
           scope = if opts[:flavor]
-            raise("Can't bulk update failed jobs") if opts[:flavor].to_s == 'failed'
-            self.scope_for_flavor(opts[:flavor], opts[:query])
+            raise("Can't #{action.to_s} failed jobs") if opts[:flavor].to_s == 'failed' && action != 'destroy'
+            scopes = []
+            scopes << self.scope_for_flavor(opts[:flavor], opts[:query])
           elsif opts[:ids]
-            self.where(:id => opts[:ids])
+            scopes = []
+            scopes << Delayed::Job.where(:id => opts[:ids])
+            failed_jobs = Delayed::Job::Failed.where(:id => opts[:ids])
+            raise("Can't #{action.to_s} failed jobs") if failed_jobs.any? && action != 'destroy'
+            scopes << failed_jobs
           end
 
           return 0 unless scope
 
           case action.to_s
           when 'hold'
-            scope.update_all(:locked_by => ON_HOLD_LOCKED_BY, :locked_at => db_time_now, :attempts => ON_HOLD_COUNT)
+            scope.map! do |x|
+              x.update_all(:locked_by => ON_HOLD_LOCKED_BY, :locked_at => db_time_now, :attempts => ON_HOLD_COUNT)
+            end
           when 'unhold'
             now = db_time_now
-            scope.update_all(["locked_by = NULL, locked_at = NULL, attempts = 0, run_at = (CASE WHEN run_at > ? THEN run_at ELSE ? END), failed_at = NULL", now, now])
+            scope.map! do |x|
+              x.update_all(["locked_by = NULL, locked_at = NULL, attempts = 0, run_at = (CASE WHEN run_at > ? THEN run_at ELSE ? END), failed_at = NULL", now, now])
+            end
           when 'destroy'
-            scope.delete_all
+            scope.map! do |x|
+              x.delete_all
+            end
           end
+          scope.flatten.reduce :+
```

Comment:
`scope` is an array of relations `[<regular jobs relation>, <failed jobs relation>]`, which when you `map!` the relations gets replaced with the number of items affected, changing it to an array like `[4, 2]`